hoomd/hpmc/pytest/test_external_field.py

```diff
@@ -45,113 +45,268 @@ def test_valid_construction_harmonicfield(device, constructor_args):
+
+        # Test that HPMCIntegrator correctly handles multiple fields.
+        field2 = hoomd.hpmc.external.Linear(default_alpha=2.345,
+                                            plane_origin=(0.1, 0.2, 0.3),
+                                            plane_normal=(-0.2, 5, -2))
+        mc.external_potentials.append(field2)
+        field2_reference_energy = 0
+        if sim.device.communicator.rank == 0:
+            for r in snapshot.particles.position:
+                field2_reference_energy += np.dot(
+                    r - field2.plane_origin,
+                    field2.plane_normal) * field2.alpha['A']
+
+        field2_energy = field2.energy
+        mc_external_energy = mc.external_energy
+
+        if sim.device.communicator.rank == 0:
+            assert field2_energy == pytest.approx(field2_reference_energy)
+            assert field_energy + field2_energy == pytest.approx(
+                mc_external_energy)
+
+    def test_normalization_of_plane_normal(self, simulation_factory,
+                                           two_particle_snapshot_factory,
+                                           add_default_integrator):
+        # create simulation & attach objects
+        sim = simulation_factory(two_particle_snapshot_factory())
+        mc, field = add_default_integrator(sim, 'linear')
+        field.plane_normal = (1, 2, 3)
+        magnitude = np.linalg.norm(field.plane_normal)
+
+        # create C++ mirror classes and set parameters
+        assert field.plane_normal == (1, 2, 3)
+        sim.run(0)
+        np.testing.assert_allclose(
+            np.array(field.plane_normal) * magnitude, (1, 2, 3))
+
+    def test_z_bias(self, device, simulation_factory, lattice_snapshot_factory):
+        """Test that particles respond to a potential as expected.
+
+        This test simulates a system of particles with a linear potential that
+        biases their z-coordinates to more negative values.
+
+        """
+        sim = simulation_factory(lattice_snapshot_factory(a=1.1, n=5))
+        mc = hoomd.hpmc.integrate.Sphere(default_d=0.01)
+        mc.shape['A'] = dict(diameter=1)
+        mc.nselect = 1
+
+        # expand box and add external field
+        old_box = sim.state.box
+        new_box = hoomd.Box(Lx=3 * old_box.Lx,
+                            Ly=3 * old_box.Ly,
+                            Lz=3 * old_box.Lz)
+        sim.state.set_box(new_box)
+        ext = hoomd.hpmc.external.Linear(default_alpha=1000,
+                                         plane_origin=(0, 0, 0),
+                                         plane_normal=(0, 0, 1))
+        mc.external_potentials = [ext]
+        sim.operations.integrator = mc
+
+        snapshot = sim.state.get_snapshot()
+        sim.run(0)
+        if snapshot.communicator.rank == 0:
+            old_mean_z = np.mean(snapshot.particles.position[:, 2])
+        old_energy = ext.energy
+
+        for n in range(10):
```

Comment:
Given the bug this test uncovered, it's a good thing you happened to realize we could adopt the test from `CPPExternalPotential`.

Reply:
I wonder if our gravity with no walls test would look more reasonable now?

Reply:
> I wonder if our gravity with no walls test would look more reasonable now?

It won't, we ran than in serial...